packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -153,32 +153,47 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
   const toggleButtonRef = React.useRef<HTMLSpanElement>(null);
   const mounted = React.useRef(false);
 
+  // Record to store all extra information (e.g. original indexes), where the key is the id of the option
+  const optionsExtraInfoRef = React.useRef<
+    Record<string, { __originalIndex: number }>
+  >({});
+
+  // Function to initialize extra info
+  const initializeExtraInfo = React.useCallback(() => {
+    optionsRef.current.forEach((option, index) => {
+      optionsExtraInfoRef.current[getOptionId(option, id)] = {
+        __originalIndex: index,
+      };
+    });
+  }, [id]);
+
+  // Latest value of the valueProp prop
```

Comment:
nit: there is no `valueProp` prop

```suggestion
  // Latest value of the value prop
```